sphinx_doc/tutorials/docs/writing_new_nav2planner_plugin.rst

```diff
@@ -0,0 +1,196 @@
+ - ``description``: Description of the plugin.
+
+.. code-block:: xml
+
+  <library path="nav2_rrtconnect_planner_plugin">
+    <class name="nav2_rrtconnect_planner/RRTConnect" type="nav2_rrtconnect_planner::RRTConnect" base_class_type="nav2_core::GlobalPlanner">
+      <description>This is an example plugin which produces global path utilizing OMPL's implemented RRT Connect Algorithm.</description>
+    </class>
+  </library>
+
+3. Next step would be to export plugin using ``CMakeLists.txt`` by using cmake function ``pluginlib_export_plugin_description_file()``. This function installs plugin description file to ``share`` directory and sets ament indexes to make it discoverable.
+
+.. code-block:: text
+
+  pluginlib_export_plugin_description_file(nav2_core global_planner_plugin.xml)
+
+4. Plugin description file should also be added to ``package.xml``
+
+.. code-block:: xml
+
+  <export>
+    <build_type>ament_cmake</build_type>
+    <nav2_core plugin="${prefix}/global_planner_plugin.xml" />
+  </export>
+
+5. Put the package in a ROS2 workspace and compile. 
+
+3- Pass the plugin name through params file
+-------------------------------------------
+
+To enable the plugin, we need to modify the ``nav2_params.yml`` file as below
+
+replace following params
+
+.. code-block:: text
+
+  planner_server:
+  ros__parameters:
+    planner_plugin_types: ["nav2_navfn_planner/NavfnPlanner"]
+    planner_plugin_ids: ["GridBased"]
+    use_sim_time: True
+    GridBased.tolerance: 2.0
+    GridBased.use_astar: false
+    GridBased.allow_unknown: true
+
+with
+
+.. code-block:: text
+
+  planner_server:
+  ros__parameters:
+    planner_plugin_types: ["nav2_rrtconnect_planner/RRTConnect"]
+    use_sim_time: True
+    GridBased.solve_time: 1.0
+    GridBased.max_trials: 4
+    GridBased.collision_checking_resolution: 0.001
+    GridBased.allow_unknown: True
+
+4- Run GradientLayer plugin
+---------------------------
+
+Run Turtlebot3 simulation with enabled navigation2. Detailed instruction how to make it are written at :ref:`getting_started`. Below is shortcut command for that:
+
+.. code-block:: bash
+
+  $ ros2 launch nav2_bringup tb3_simulation_launch.py params_file:=/path/to/your_params_file.yaml
+
+Then goto RViz and click on the "2D Pose Estimate" button at the top and point the location on map as it was described in :ref:`getting_started`. Robot will localize on the map and then click on "Navigation2 goal" and click on the pose where you want your planner to consider a goal pose. After that planner will plan the path and robot will start moving towards the goal. Please checkout the video below for demo.
+
+.. raw:: html
+
+    <h1 align="center">
+      <div>
+        <div style="position: relative; padding-bottom: 0%; overflow: hidden; max-width: 100%; height: auto;">
+          <iframe width="450" height="300" src="https://www.youtube.com/embed/MSy7WAE_xz4?autoplay=1&mute=1" frameborder="1" allowfullscreen></iframe>
```

Comment:
Some of these paths are _really_ bad -  I think we need to do some work on the planner itself.

I think there's 2 ways to make the tutorial. A toy example (straight line planner, something unrealistic for actual use so no one actually tries to) or a working example (you can use it if you want since it works, but simple). We need to find a balance to put this in. I think it would look really bad on ROS2 Navigation if folks wanted to use a sampling based planner and saw these results as the "out of the box" experience with them.

Reply:
If you'd like, we could make a straight line planner as the tutorial and work together on refining this OMPL work into a supported planner in navigation.

Reply:
I don't mind if you are fine with it. I was thinking of making tutorials with some thing concrete where the user gets something that will work in the end.